libopenage/util/fslike/directory.cpp

```diff
@@ -64,22 +65,22 @@ std::string Directory::resolve(const Path::parts_t &parts) const {
 }
 
 
-// TODO: use std::optional when available
-std::tuple<struct stat, int> Directory::do_stat(const Path::parts_t &parts) const {
+/* Added the usage of std::optional here together with std::filesystem (a.k.a fs)
+   currently only do_stat, is_file and is_dir updated*/
+std::optional<fs::file_status> Directory::do_stat(const Path::parts_t &parts) const {
 	const std::string path = this->resolve(parts);
-	struct stat buf;
-	int result = stat(path.c_str(), &buf);
-
-	return std::make_tuple(buf, result);
+	fs::path fsPath{path};
```

Comment:
if you didn't have a look at [our codestyle](https://github.com/SFTtech/openage/blob/master/doc/code_style/) yet, we write variables `in_snake_case`, so that should be `fs_path`.

Reply:
You know when you read something and it goes in and then straight out? Yeah, sorted in the next commit, thank you!